src/bitcoind_client.rs

```diff
@@ -254,7 +254,7 @@ impl BroadcasterInterface for BitcoindClient {
 			let mut rpc = bitcoind_rpc_client.lock().await;
 			// This may error due to RL calling `broadcast_transaction` with the same transaction
 			// multiple times, but the error is safe to ignore.
-			match rpc.call_method::<RawTx>("sendrawtransaction", &vec![tx_serialized]).await {
+			match rpc.call_method::<TxidHex>("sendrawtransaction", &vec![tx_serialized]).await {
```

Comment:
Maybe we should parse this as a `Txid`.

https://docs.rs/bitcoin/0.27.0/bitcoin/hash_types/struct.Txid.html

Reply:
I tried that, but I couldn't implement `TryInto<Txid> for JsonResponse` since `Txid` is not defined in the current crate.

I may be missing something though.

Reply:
Ah, right... 😕 Let's add that to `lighting-block-sync::convert` since it is a common enough type. Should be able to do it there like we do for `Block`:

https://github.com/rust-bitcoin/rust-lightning/blob/69ee4860848d5992b238eca3343141004d9d1572/lightning-block-sync/src/convert.rs#L109

Reply:
Oh wait, is the other way around isn't it? The reason is because `JsonResponse` is not defined in the current crate, not `Txid` 😅 .

Anyway, I'll define it in `lighting-block-sync::convert`

Reply:
Would it make sense to also make `BitcoindClient::sign_raw_transaction_with_wallet` return `bitcoin::Transaction` instead of the `SignedTx` wrapper and therefore also add `TryInto<Transaction> for JsonResponse` to `lighting-block-sync::convert`?

I'm wondering given `ldk-sample` asserts the transaction has been properly signed and the casts it to `bitcoin:Transaction` (https://github.com/lightningdevkit/ldk-sample/blob/main/src/main.rs#L136-L139), which can be made part of `sign_raw_transaction_with_wallet` and return `(Transaction, bool)` or `Result<Transaction, E>`.

If so, then `send_raw_transaction` could simply receive a `bitcoin::Transaction` and we could get rid of the `SignedTx` wrapper altogether.

Reply:
SGTM. Would prefer a `Result`. Bonus if the error type could contain the relevant data parsed into their respective types.